testutil/daemon/daemon.go

```diff
@@ -217,6 +217,32 @@ func New(t testing.TB, ops ...Option) *Daemon {
 	return d
 }
 
+// BinaryPath returns the binary and its arguments.
+func (d *Daemon) BinaryPath() (string, []string, error) {
+	dockerdBinary, err := exec.LookPath(d.dockerdBinary)
+	if err != nil {
+		return "", nil, errors.Wrapf(err, "[%s] could not find docker binary in $PATH", d.id)
+	}
```

Comment:
Perhaps we should keep it simple and only do this part here, and keep the special handling for rootless in `StartWithLogFile()`